Taken from Oracle solaris-userland b83a793,
components/x11/xserver/xorg/patches/02-sun-paths.patch (xorg-server 21.1.24).

Solaris specific patches for paths and manpages

Not suitable for upstream

Existing set of patches from the X gate

--- a/hw/xfree86/common/xf86Config.c	2019-02-12 09:53:14.207906190 -0800
+++ b/hw/xfree86/common/xf86Config.c	2019-02-12 10:16:15.616458998 -0800
@@ -75,9 +75,9 @@
  */
 #ifndef ALL_CONFIGPATH
 #define ALL_CONFIGPATH	"%A," "%R," \
-			"/etc/X11/%R," "%P/etc/X11/%R," \
+			"/etc/X11/%R," "%P/lib/X11/%R," \
 			"%E," "%F," \
-			"/etc/X11/%F," "%P/etc/X11/%F," \
+			"/etc/X11/%F," "%P/lib/X11/%F," \
 			"/etc/X11/%X," "/etc/%X," \
 			"%P/etc/X11/%X.%H," \
 			"%P/etc/X11/%X," \
@@ -85,8 +85,8 @@
 			"%P/lib/X11/%X"
 #endif
 #ifndef RESTRICTED_CONFIGPATH
-#define RESTRICTED_CONFIGPATH	"/etc/X11/%S," "%P/etc/X11/%S," \
-			"/etc/X11/%G," "%P/etc/X11/%G," \
+#define RESTRICTED_CONFIGPATH	"/etc/X11/%S," "%P/lib/X11/%S," \
+			"/etc/X11/%G," "%P/lib/X11/%G," \
 			"/etc/X11/%X," "/etc/%X," \
 			"%P/etc/X11/%X.%H," \
 			"%P/etc/X11/%X," \
--- a/hw/xfree86/common/xf86Globals.c	2019-02-12 10:17:05.369152003 -0800
+++ b/hw/xfree86/common/xf86Globals.c	2019-02-12 10:18:05.887684546 -0800
@@ -137,7 +137,8 @@
 
 const char *xf86ConfigFile = NULL;
 const char *xf86ConfigDir = NULL;
-const char *xf86ModulePath = DEFAULT_MODULE_PATH;
+/* Sun: add old module path for compatibility with out-of-tree driver packages */
+const char *xf86ModulePath = DEFAULT_MODULE_PATH ",/usr/X11/lib/modules/";
 MessageType xf86ModPathFrom = X_DEFAULT;
 const char *xf86LogFile = DEFAULT_LOGDIR "/" DEFAULT_LOGPREFIX;
 MessageType xf86LogFileFrom = X_DEFAULT;
--- a/hw/xfree86/man/xorg.conf.man	2019-02-12 10:18:24.417826745 -0800
+++ b/hw/xfree86/man/xorg.conf.man	2019-02-12 10:19:29.947011371 -0800
@@ -36,9 +36,9 @@
 .RS 4
 .nf
 .IR /etc/X11/ <cmdline>
-.IR @projectroot@/etc/X11/ <cmdline>
+.IR @projectroot@/lib/X11/ <cmdline>
 .IB /etc/X11/ $XORGCONFIG
-.IB @projectroot@/etc/X11/ $XORGCONFIG
+.IB @projectroot@/lib/X11/ $XORGCONFIG
 .I /etc/X11/xorg.conf
 .I /etc/xorg.conf
 .IR @projectroot@/etc/X11/xorg.conf. <hostname>
@@ -67,10 +67,10 @@
 .nf
 <cmdline>
 .IR /etc/X11/ <cmdline>
-.IR @projectroot@/etc/X11/ <cmdline>
+.IR @projectroot@/lib/X11/ <cmdline>
 .B $XORGCONFIG
 .IB /etc/X11/ $XORGCONFIG
-.IB @projectroot@/etc/X11/ $XORGCONFIG
+.IB @projectroot@/lib/X11/ $XORGCONFIG
 .I /etc/X11/xorg.conf
 .I /etc/xorg.conf
 .IR @projectroot@/etc/X11/xorg.conf. <hostname>
--- a/include/xorg-server.h.in	2019-02-12 10:19:47.220370143 -0800
+++ b/include/xorg-server.h.in	2019-02-12 10:20:09.366590998 -0800
@@ -203,6 +203,7 @@
 /* Have X server platform bus support */
 #undef XSERVER_PLATFORM_BUS
 
+#include <sys/isa_defs.h> /* Ensure _LP64 is defined when needed on Solaris */
 #ifdef _LP64
 #define _XSERVER64 1
 #endif
